airflow/providers/google/cloud/hooks/vertex_ai/hyperparameter_tuning_job.py

```diff
@@ -0,0 +1,415 @@
+        custom_job: CustomJob,
+        metric_spec: Dict[str, str],
+        parameter_spec: Dict[str, hyperparameter_tuning._ParameterSpec],
+        max_trial_count: int,
+        parallel_trial_count: int,
+        max_failed_trial_count: int = 0,
+        search_algorithm: Optional[str] = None,
+        measurement_selection: Optional[str] = "best",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        labels: Optional[Dict[str, str]] = None,
+        encryption_spec_key_name: Optional[str] = None,
+    ) -> HyperparameterTuningJob:
+        """Returns HyperparameterTuningJob object"""
+        return HyperparameterTuningJob(
+            display_name=display_name,
+            custom_job=custom_job,
+            metric_spec=metric_spec,
+            parameter_spec=parameter_spec,
+            max_trial_count=max_trial_count,
+            parallel_trial_count=parallel_trial_count,
+            max_failed_trial_count=max_failed_trial_count,
+            search_algorithm=search_algorithm,
+            measurement_selection=measurement_selection,
+            project=project,
+            location=location,
+            credentials=self._get_credentials(),
+            labels=labels,
+            encryption_spec_key_name=encryption_spec_key_name,
+        )
+
+    def get_custom_job_object(
+        self,
+        display_name: str,
+        worker_pool_specs: Union[List[Dict], List[gapic.WorkerPoolSpec]],
+        base_output_dir: Optional[str] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        labels: Optional[Dict[str, str]] = None,
+        encryption_spec_key_name: Optional[str] = None,
+        staging_bucket: Optional[str] = None,
+    ) -> CustomJob:
+        """Returns CustomJob object"""
+        return CustomJob(
+            display_name=display_name,
+            worker_pool_specs=worker_pool_specs,
+            base_output_dir=base_output_dir,
+            project=project,
+            location=location,
+            credentials=self._get_credentials,
+            labels=labels,
+            encryption_spec_key_name=encryption_spec_key_name,
+            staging_bucket=staging_bucket,
+        )
+
+    @staticmethod
+    def extract_hyperparameter_tuning_job_id(obj: Dict) -> str:
+        """Returns unique id of the hyperparameter_tuning_job."""
+        return obj["name"].rpartition("/")[-1]
+
+    def wait_for_operation(self, operation: Operation, timeout: Optional[float] = None):
+        """Waits for long-lasting operation to complete."""
+        try:
+            return operation.result(timeout=timeout)
+        except Exception:
+            error = operation.exception(timeout=timeout)
+            raise AirflowException(error)
+
+    def cancel_hyperparameter_tuning_job(self) -> None:
+        """Cancel HyperparameterTuningJob"""
+        if self._hyperparameter_tuning_job:
+            self._hyperparameter_tuning_job.cancel()
```

Comment:
NIT: maybe we could consider to create generic methods, logic look the same between hooks